matrix_client/room.py

```diff
@@ -468,14 +500,18 @@ def get_joined_members(self):
         Returns:
             {user_id: {"displayname": str or None}}: Dictionary of joined members.
         """
+        if self._members:
+            return self._members
         response = self.client.api.get_room_members(self.room_id)
-        rtn = {
-            event["state_key"]: {
-                "displayname": event["content"].get("displayname"),
-            } for event in response["chunk"] if event["content"]["membership"] == "join"
-        }
-
-        return rtn
+        for event in response["chunk"]:
+            if event["content"]["membership"] == "join":
+                self._mkmembers({
+                    "user_id": event["state_key"],
+                    "displayname": event["content"].get("displayname")})
+        return self._members
+
+    def _mkmembers(self, member):
```

Comment:
Not sure I understand the utility of this method. Why not just call
`self._members.append` directly?

Reply:
I used the same approach as in `client._mkroom` for testing.